### PR TITLE
Adds a config option to disable the security cyborg module.

### PR DESCRIPTION
**Pull request by Shadowlight213**
Requested by @hornygranny 
Default is to allow the module.

### CODE REVIEW

**Comment by PKPenguin321** [comment]
i'm ready

**Comment by Aloraydrel** [comment]
Remove meme borg 2016

**Comment by Dawgas** [comment]
personally don't get why we're allowing the whining of more than a half decade of small children to influence the game

might as well also include an option to not allow cyborgs to be made at all, and be more modular with cyborg types, so "we" can plug and pray new borg modules at will

ps: secborg's been nerfed more than enough, this is too far

**Comment by Shadowlight213** [comment]
@Dawgas sure. I can make a disallow borg config next. HG wanted this made so we can run a week with no secborg test.

**Comment by PKPenguin321** [comment]
@Dawgas 
personally don't get why we're allowing the whining of more than a half decade of small children to influence the game

we should have removed secborgs years ago

**Comment by xxalpha** [comment]
So you remove the only combat cyborg.

Don't open ban requests when a cyborg ignores your orders to go fight something.

**Comment by PKPenguin321** [comment]
don't worry, i wont

**Comment by Zilenan91** [comment]
We don't allow cowards in this spess

**Comment by KorPhaeron** [comment]
Secborgs are good as a murder spree deterent.

I dont really get the hate for the security module in particular. Might put me in the minority.

>its a config option/temporary test

I've used this trick too many times to fall for it

**Comment by Zilenan91** [comment]
The reason I dislike secborgs is because it's just way too good for everything imaginable. 

For antagonists in particular, if they emag it, they get an unstunnable, undisarmable infinite ammo laser gun on wheels that also has a disabler and a stunbaton, so effectively infinite ammo ranged and melee stuns that can only effectively be fought by security.

For everyone else, dealing with secborgs just flat out isn't fun. Even as a human they'll just say fuck you harm lelelelelelele and brig you for 5 minutes for doing the least harmful things imaginable, and as a non-human you best have a metafriend in security because they're going to space you just because they can, all because they're the only borg that has the ability to detain people.

They're also an all-access valid machine who has a license to griff non-humans and tremendously fuck with humans so people play secborgs, shit on people, and don't get punished for it. Some of the worst players I've ever encountered have been secborgs, it's grief lottery : the job except you can grief people every round.

**Comment by PKPenguin321** [comment]
>zilenan knowing what he's talking about for once

it's a whole new wooooorld

**Comment by Core0verload** [comment]
It isn't uncommon to see a glorious emagged/malf secborg getting it's power cell raped by a random assistant with a crowbar and a stolen flash. And, AFAIK, one well thrown flashbang can give secs enough time to approach it, flash it again and remove the cell. 

Shit players playing as secborgs are still a trouble, but it seems to be an administrative problem.

**Comment by xxalpha** [comment]
They won't use the flashbang argument here, only in the nerf PR.

**Comment by duncathan** [comment]
To everyone disagreeing with this, make sure to answer this question: What does secborg add to the game to counteract the things it detracts from the game?

If you can't answer that you're gonna be completely overlooked

I'm neutral on the issue but I wanna help both sides make their points

**Comment by xxalpha** [comment]
Here's simple logic for you:

AI is Asimov.
Asimov's priority law is not causing harm and preventing harm.
Only one cyborg module is equipped with pacifying tools to keep humans from harming each other.

You want to remove secborg? Go ahead, but don't cry at Asimov silicons not interfering with harmful conflict, because they can't really stop it without secborg.

**Comment by Cebasiz** [comment]
>secborgs are too strong

maybe like a year and a half ago

now like half the shit on the station is instant GGNORE. if you are hit by 

-ion rifle
-EMP grenade
-flash
-laser pointer
flashbang (full screen like 10+ second stun)
EMP flashlight
EMP implant
any one of a dozen antagonist dedicated cyborg fuck you(s)

you can be beaten to death by someone, anyone with a crowbar

the borg serves it's purpose, to prevent harm, which is what the server's DEFAULT LAWSET is all about. it makes absolutely no goddamn sense to remove the only cyborg who's even capable of doing the thing that the entire lawset is built around and jam the rest of the borgs into that niche which the secborg ALREADY FILLS

**Comment by Zilenan91** [comment]
Macrobombs served their function, making big explosions. The reason they were removed is because they weren't fun, exactly like secborgs. It doesn't matter at all how efficient or how good they are if they flat out aren't fun for anyone else to deal with.

**Comment by Cheridan** [comment]
@xxalpha You want to remove secborg? Go ahead, but don't cry at Asimov silicons not interfering with harmful conflict, because they can't really stop it without secborg.

That's literally the point of why secborg is garbage. The AI is supposed to inform the human crew where harm is happening and then the crew handles the situation. It's not meant to neutralize all conflict by itself while twiddles and wishes something interesting was happening.

**Comment by Cebasiz** [comment]
it doesn't actually matter because you aren't even solving the problem you're trying to solve.

all that's going to happen is you're going to force every other borg type to badly try to do the same job

but whatever, nobody actually wants to solve these problems, they just want to have a witch hunt

whatever

**Comment by PKPenguin321** [comment]
>all that's going to happen is you're going to force every other borg type to badly try to do the same job

1. Remove inaction clause from law 1
2. Borg tells crew where harm is occurring and who is committing it
3. Borg leaves

**Comment by Zilenan91** [comment]
Or keep it and let the borg throw itself at the enemy like it's supposed to.

**Comment by xxalpha** [comment]
>Remove inaction clause from law 1

I am all for that, to be quite honest with you my man.

**Comment by Cebasiz** [comment]
i suppose this is progressing from the discussion recently, however i'm not really a fan. the inaction clause is a good time. it makes you be a nice guy and put people out when they're on fire, or stop them from bleeding to death or whatever.

what is the problem removing it even solves? antag hunting is no longer 'preventing harm' so it's illegalP? why not just jump directly to saying that and keep the laws intact?

besides, sillicons should be helping the crew against the nastier antags like shadowlings, changelings etc. so changing the lawset so they contentedly let the crew die in combat against inhuman monsters seems a little silly, not to mention the can of worms this will open regarding what exactly qualifies as not committing harm

tl;dr this seems like a semi-pointless half measure. if you want the sillicons to be uninvolved you should just give them drone laws. if you want them to be involved but not validhunt why don't you just ban validhunting?

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 1]
```diff
@@ -222,6 +226,9 @@
 			feedback_inc("cyborg_medical",1)
 
 		if("Security")
+			if(!config.allow_secborg)
+				src << "spam class = 'warning'>ERROR: NANOTRASEN HAS DISABLED THIS MODULE. PLEASE SELECT A DIFFERENT ONE!</span>"
```
Comment (tkdrg) [inline code/modules/mob/living/silicon/robot/robot.dm]: typo
shoul be `<span class = 'warning'>` instead

**Comment by tkdrg** [comment]
secborg players are cancer, also it's just a config option

File: code/modules/mob/living/silicon/robot/robot.dm (hunk 1 above)
Comment (Shadowlight213) [inline code/modules/mob/living/silicon/robot/robot.dm]: >shoul

Type in a note about a typo. How meta.

Comment (RemieRichards) [inline code/modules/mob/living/silicon/robot/robot.dm]: Why even have this section.
you can't pick it if it's not in the list.

this can't even be spoofed.

Comment (Shadowlight213) [inline code/modules/mob/living/silicon/robot/robot.dm]: RemieRichards I actually coded that first before removing it from the list. I'll remove it.

**Comment by Aloraydrel** [comment]
JUST DO ITTTTTTTT

**Comment by KorPhaeron** [comment]
Fix conflicts

**Comment by PKPenguin321** [comment]
Do it

**Comment by Zilenan91** [comment]
IT'S TIIIIIIME

File: code/controllers/configuration.dm [hunk 2]
```diff
@@ -517,6 +518,8 @@
 					config.force_random_names		= 1
 				if("allow_ai")
 					config.allow_ai					= 1
+				if("disable_secborg")
+					config.allow_secborg			= 0
```
Comment (KorPhaeron) [inline code/controllers/configuration.dm]: You did this backwards

Comment (xxalpha) [inline code/controllers/configuration.dm]: How is it backwards?

Comment (Shadowlight213) [inline code/controllers/configuration.dm]: It's not backwards. The config defaults to having secborg be on.

**Comment by Shadowlight213** [comment]
@tkdrg conflicts resolved

File: code/controllers/configuration.dm (hunk 2 above)
Comment (Jordie0608) [inline code/controllers/configuration.dm]: Would be better if config name was consistent, instead of changing from disable_secborg to allow_secborg

Comment (Shadowlight213) [inline code/controllers/configuration.dm]: K

**Comment by Shadowlight213** [comment]
@Jordie @KorPhaeron fixed.
